src/util/validate_expressions.cpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************\
+
+Module: Validate expressions
+
+Author: Daniel Poetzl
+
+\*******************************************************************/
+
+#include "validate_expressions.h"
+
+#include "expr.h"
+#include "namespace.h"
+#include "std_expr.h"
+#include "validate.h"
+
+#define CALL_ON_EXPR(expr_type)                                                \
+  C<expr_type>()(expr, std::forward<Args>(args)...)
+
+template <template <typename> class C, typename... Args>
+void call_on_expr(const exprt &expr, Args &&... args)
+{
+  if(expr.id() == ID_equal)
+  {
+    CALL_ON_EXPR(equal_exprt);
+  }
+  else if(expr.id() == ID_plus)
+  {
+    CALL_ON_EXPR(plus_exprt);
+  }
+  else
+  {
+    CALL_ON_EXPR(exprt);
+  }
```

Comment:
More cases will be added here in the future.

Reply:
I've added a macro `REPORT_UNIMPLEMENTED_EXPRESSION_CHECKS` that can be set when compiling. If enabled the ids for currently unimplemented expression checks are reported to `stderr`.